### ISSUE TITLE
(not an issue, help request)

### ISSUE THREAD

**Issue by noreplacementfordisplacement** [OrigamiSL/FDNet2023#6]
I used a custom format and modified the exchange data in the file as below:

data_parser = {
    **'Exchange': {'data': 'Exchange_reference.csv', 'M': [1, 1], 'S': [1, 1]},**

When I run the file as following:

python -u main_reference.py --model FDNet --data Exchange --root_path ./data/Exchange/ --features S  --label_len 
96  --pred_len 96 --pyramid 0 --attn_nums 1 --kernel 3 --learning_rate 0.0001 --dropout 0.1 --criterion Standard --d_model 8 --timebed None --batch_size 16 --itr 10 --trai

I continuously run into the following test / val / train sizes. I do not understand why this is the case? What determines the train / val test size?

![afbeelding](https://github.com/OrigamiSL/FDNet-KBS-2023/assets/18721012/8f39cd50-dcb6-46cc-a284-ec1d0e4ab757)

**Comment by OrigamiSL** [comment]
The sizes of train/val/test subsets are pre-determined and fixed in the ./data/dataloader.py. You can easily modify them in the corresponding dataloader functions, e.g., Dataset_Custom (Line293-Line299 in ./data/dataloader.py). They are fixed since these settings are commonly used in the deep time-series forecasting community.

**Comment by noreplacementfordisplacement** [comment]
Thanks, I will try it! 

I have another question (mainly regarding hyperparameter optimization). I am working with several univariate datasets, for which I modify the bash command running the **exchange** experiment.

1) Are there any parameters which I need to configure manually to deal with a univariate dataset as below (example)?


<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/dhoppene/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/dhoppene/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">


date | 0
-- | --
1/1/1990 | 123.45
2/1/1990 | 123.33
3/1/1990 | 123.11
4/1/1990 | 123.22
5/1/1990 | 124.01
6/1/1990 | 123.99



</body>

</html>

**Comment by OrigamiSL** [comment]
1.The formula of the date time if you want to use the time embedding features.
2.The name of the variable, e.g., your provided name '0'; You shall designate the value of '--target' as the variable name or manually rename your variable name in the dataset to be 'OT' as it is the default value of '--target' argument.

**Comment by noreplacementfordisplacement** [comment]
Thank you for the quick response.

For 1. What time formats are supported? How does the time format affect the result? For example let's say I want to use millisecond data
For 2. I modified the header to say OT. In the exchange dataset there are multiple columns. Is OT the output data and are the other columns multivariate input data? (example below)


<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/dhoppene/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/dhoppene/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">


date | 0 | 1 | 2 | 3 | 4 | 5 | 6 | OT
-- | -- | -- | -- | -- | -- | -- | -- | --
1/1/1990 0:00 | 0.7855 | 1.611 | 0.861698 | 0.634196 | 0.211242 | 0.006838 | 0.525486 | 0.593
1/2/1990 0:00 | 0.7818 | 1.61 | 0.861104 | 0.633513 | 0.211242 | 0.006863 | 0.523972 | 0.594
1/3/1990 0:00 | 0.7867 | 1.6293 | 0.86103 | 0.648508 | 0.211242 | 0.006975 | 0.526316 | 0.5973
1/4/1990 0:00 | 0.786 | 1.637 | 0.862069 | 0.650618 | 0.211242 | 0.006953 | 0.523834 | 0.597
1/5/1990 0:00 | 0.7849 | 1.653 | 0.861995 | 0.656254 | 0.211242 | 0.00694 | 0.527426 | 0.5985



</body>

</html>


Another question: do you have guidelines on how hyperparameters affect the result? E.g. batch size, d_model, iterations, learning rate, label_len? Do you have a good resource to understand these hyperparameters?

**Comment by OrigamiSL** [comment]
I'm afraid that you had better learn some prerequisite basics about deep learning, neural network and deep time-series forecasting. Otherwise, you will have tons of questions that can overwhelm me. Afterwards, a careful reading to README.MD and my paper is also suggested to avoid some elementary questions.

**Comment by noreplacementfordisplacement** [comment]
Thanks. If you have a tip, for a recommended resource to get me up to speed quickly it helps (book / youtube video / tutorial).
